src/Dispatch.purs

```diff
@@ -51,23 +89,40 @@ data VisibilityStatus
   | Off
   | Toggle
 
-data Item item
-  = Selected item
-  | Selectable item
-  | Disabled item
-
-instance showItem :: Show item => Show (Item item) where
-  show (Selected item)   = "Selected: " <> show item
-  show (Selectable item) = "Selectable: " <> show item
-  show (Disabled item)   = "Disabled: " <> show item
-
-instance eqItem :: Eq item => Eq (Item item) where
-  eq (Selected a) (Selected b) = eq a b
-  eq (Selectable a) (Selectable b) = eq a b
-  eq (Disabled a) (Disabled b) = eq a b
-  eq (Selected a) _ = false
-  eq (Selectable a) _ = false
-  eq (Disabled a) _ = false
+type ContainerState item =
+  { items            :: Array item
+  , open             :: Boolean
+  , highlightedIndex :: Maybe Int
+  , lastIndex        :: Int
+  , mouseDown        :: Boolean
+  }
+
+type ContainerInput item o e =
+  { items  :: Array item
+  , render :: ContainerState item -> H.ComponentHTML (Dispatch item o e) }
+
+
+emit :: ∀ a0 a1 o item e f. Applicative f => (o Unit -> f Unit) -> Dispatch item o e a0 -> a1 -> f a1
+emit f (ParentQuery o _) a = a <$ f o
+emit _ _ a = pure a
+
+updateState :: ∀ state html. (state -> state) -> Store state html -> Store state html
+updateState f inputStore =
+  let (Tuple r oldState) = runStore inputStore
+   in store r $ f oldState
```

Comment:
Is this just `seeks`? https://pursuit.purescript.org/packages/purescript-transformers/3.5.0/docs/Control.Comonad.Store#v:seeks

Reply:
You're right! Updated in latest commit. Thanks for the note!